Clover/app/src/main/java/org/floens/chan/core/site/parser/CommentParser.java

```diff
@@ -233,6 +235,12 @@ public CharSequence handleTable(Theme theme,
 
     public Link matchAnchor(Post.Builder post, CharSequence text, Element anchor, PostParser.Callback callback) {
         String href = anchor.attr("href");
+        // gets us something like /board/thread/postno#quoteno
+        //hacky fix for 4chan having two domains but the same API
+        if(href.matches("//boards\\.4chan.*?\\.org/(.*?)/thread/(\\d*?)#p(\\d*)")) {
+            href = href.substring(2);
+            href = href.substring(href.indexOf('/'));
```

Comment:
instead of string manipulation, could you use the [Uri class](https://developer.android.com/reference/android/net/Uri.html)?

Reply:
The rest of the method expects something in the form of `(thread no)#p(post no)` and does regex matching against that; this code gets the string value to the correct form as quickly as possible. It's possible that the entire method may be redone with URI's, however I don't know enough about the architecture of the project to make that call.